frontend/src/components/TransactionsList.tsx

```diff
@@ -1,12 +1,129 @@
-import { DrumIcon } from "lucide-react";
+import { DownloadIcon, DrumIcon, MoreHorizontalIcon } from "lucide-react";
 import { useRef, useState } from "react";
 import { CustomPagination } from "src/components/CustomPagination";
 import EmptyState from "src/components/EmptyState";
 import Loading from "src/components/Loading";
 import TransactionItem from "src/components/TransactionItem";
-
+import { Button } from "src/components/ui/button";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "src/components/ui/dropdown-menu";
+import { UpgradeDialog } from "src/components/UpgradeDialog";
 import { LIST_TRANSACTIONS_LIMIT } from "src/constants";
+import { useAlbyMe } from "src/hooks/useAlbyMe";
 import { useTransactions } from "src/hooks/useTransactions";
+import { ListTransactionsResponse, Transaction } from "src/types";
+import { request } from "src/utils/request";
+
+const convertToCSV = (transactions: Transaction[]) => {
+  if (!transactions.length) {
+    return "";
+  }
+
+  // Get headers from the first transaction
+  const headers = Object.keys(transactions[0]);
+  const csvHeaders = headers.join(",");
+
+  // Convert each transaction to CSV row
+  const csvRows = transactions.map((tx) => {
+    return headers
+      .map((header) => {
+        const value = (tx as Record<string, unknown>)[header];
+        // Escape commas and quotes in values
+        if (
+          typeof value === "string" &&
+          (value.includes(",") || value.includes('"'))
+        ) {
+          return `"${value.replace(/"/g, '""')}"`;
+        }
+        return value;
+      })
+      .join(",");
+  });
+
+  return [csvHeaders, ...csvRows].join("\n");
+};
+
+const handleDownloadTransactions = async (appId?: number) => {
+  try {
+    // Fetch all transactions by paginating through all pages
+    let allTransactions: Transaction[] = [];
+    let offset = 0;
+    let hasMoreTransactions = true;
+
+    while (hasMoreTransactions) {
+      let url = `/api/transactions?limit=${LIST_TRANSACTIONS_LIMIT}&offset=${offset}`;
+      if (appId) {
+        url += `&appId=${appId}`;
+      }
+
+      const data = (await request(url)) as ListTransactionsResponse;
```

Comment:
can you check how we do this elsewhere (don't use brackets around the await, and throw an exception if the response is not truthy)